layers.py

```diff
@@ -300,7 +301,7 @@ def __init__(self, d_model, num_heads, d_ff, max_rel_dist, bias=True, dropout=0.
         self.num_heads = num_heads
         self.max_rel_idst = max_rel_dist
 
-        self.mha = MultiHeadAttention(d_model, num_heads, max_rel_dist, bias)
+        self.self_attn = MultiHeadAttention(d_model, num_heads, max_rel_dist, bias)
```

Comment:
I would like this to still be left as `self.mha`. This is because it is not necessarily computing self attention, and in general is just computing MultiHeadAttention. One can, in principle, add an encoder on top of this decoder so that it would not be computing just self attention

Reply:
This was changed because there is a crash in the pytorch 2, this line fails, https://github.com/pytorch/pytorch/blob/07cecf4168503a5b3defef9b2ecaeb3e075f4761/torch/nn/modules/transformer.py#L490 I guess you could have a dummy self_attn block with batch_first=False but I would have to test if it causes other issues.

Reply:
I see, alright I'll leave your change that way.